www/netj00/HW10/js/main.js

```diff
@@ -0,0 +1,151 @@
+    "Rammus", "RekSai", "Renekton", "Rengar", "Riven", "Rumble", "Ryze", "Sejuani", "Shaco", "Shen", "Shyvana", "Singed",
+    "Sion", "Sivir", "Skarner", "Sona", "Soraka", "Swain", "Syndra", "TahmKench", "Taliyah", "Talon", "Taric", "Teemo",
+    "Thresh", "Tristana", "Trundle", "Tryndamere", "TwistedFate", "Twitch", "Udyr", "Urgot", "Varus", "Vayne", "Veigar",
+    "Velkoz", "Vi", "Viktor", "Vladimir", "Volibear", "Warwick", "Xayah", "Xerath", "XinZhao", "Yasuo", "Yorick", "Zac",
+    "Zed", "Ziggs", "Zilean", "Zoe", "Zyra"];
+
+var cards = [];
+var eGame = document.getElementById("game-field");
+var ePoints = document.getElementById("points");
+var flippedCard1;
+var flippedCard2;
+var score = 0;
+
+// --------------------------------------------------------------------------------------------------------------------
+
+
+function populateCards() {
+    let tempChampions = champions.slice(0);
+    for (let n = 0; n <= 9; n++) {
+        let random = Math.floor(Math.random() * tempChampions.length);
+        cards.push(tempChampions[random]);
+        cards.push(tempChampions[random]);
+        tempChampions.splice(random, 1);
+    }
+}
+
+function shuffleArray(array) {
+    return array.sort(function () {
+        return 0.5 - Math.random();
+    });
+}
+
+function createCardElement(id) {
+    let elementGame = document.createElement("div");
+    elementGame.classList.add("card");
+    elementGame.id = id;
+    elementGame.addEventListener("click", cardsClick);
+    elementGame.addEventListener("dblclick", cardsClick);
+    eGame.appendChild(elementGame);
+}
+
+function cardsClick(e) {
+    if (e.target.classList.contains("revealed"))
+        return;
+    if (flippedCard1 && flippedCard2)
+        compareCards();
+    if (flippedCard1) {
+        flippedCard2 = e.target;
+        hadWon();
+    }
+    else
+        flippedCard1 = e.target;
+    e.target.innerText = cards[e.target.id];
+    e.target.classList.add("revealed");
+}
+
+function populateGameField() {
+    for (let i = 0; i < cards.length; i++) {
+        createCardElement(i);
+    }
+}
+
+function compareCards() {
+    if (flippedCard1.innerText !== flippedCard2.innerText) {
+        flippedCard1.innerText = "";
+        flippedCard2.innerText = "";
+        flippedCard1.classList.remove("revealed");
+        flippedCard2.classList.remove("revealed");
+    }
+    else {
+        score++;
+        ePoints.innerText = score;
+    }
+    flippedCard1 = null;
+    flippedCard2 = null;
+}
+
+function hadWon() {
+    let revealedCards = eGame.getElementsByClassName("revealed");
+    console.log(revealedCards.length);
+    if (revealedCards.length >= 19) {
+        score++;
+        alert("Congratulations! You have won some weird pexeso... but was it worth your time? Anyway here is your score: " + score);
+    }
+}
+
+
+// -------------------------------------------main---------------------------------------------------------------------
+
+champions = shuffleArray(champions);
```

Comment:
nwm, moc se mi nezda, nejak mockrat michas karty, uz i pri  vyberu jmen, uplne staci to udelat jednou